core/handler/handler.go

```diff
@@ -170,10 +171,8 @@ func (h *handler) Init(c *component.Component) error {
 
 	h.Component.SetStatus(component.StatusHealthy)
 	if h.Component.Monitor != nil {
-		h.monitorStream = h.Component.Monitor.NewHandlerStreams(h.Identity.Id, h.AccessToken)
-		go h.Component.Monitor.TickStatus(func() {
-			h.monitorStream.Send(h.GetStatus())
-		})
+		h.monitorStream = h.Component.Monitor.BrokerClient(h.Context, grpc.PerRPCCredentials(auth.WithStaticToken(h.AccessToken)))
+		go h.Component.Monitor.TickStatus(func() { h.monitorStream.Send(h.GetStatus()) })
```

Comment:
this goroutine never returns. Which may cause trouble. What if the component is down/disconnected? Do we need to query it's status all the time and send it?